Application/components/FriendsColumn/FriendRequestManager.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useEffect, FC } from 'react';
+import { Menu, Divider, Text, Notification, Button } from '@mantine/core';
+import { useChannel } from "ably/react";
+import { getSystemsChannelID} from "@/utility";
+
+interface FriendRequest {
+  id: string;
+  username: string;
+}
+
+interface FriendRequestManagerProps {
+  userId: string;
+}
+
+const FriendRequestManager: FC<FriendRequestManagerProps> = ({ userId }) => {
+  const [sentRequests, setSentRequests] = useState<FriendRequest[]>([]);
+  const [receivedRequests, setReceivedRequests] = useState<FriendRequest[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const { channel, ably } = useChannel(getSystemsChannelID());
+
+  const fetchFriendRequests = async () => {
+    setLoading(true);
+    try {
```

Comment:
-  Ensure proper error handling throughout the component. Currently, errors are caught and logged, but there is no user feedback or indication of what went wrong. Consider displaying error messages to the user or implementing a retry mechanism for failed requests.
- The component polls for friend requests every 5 seconds (setInterval(fetchFriendRequests, 5000)). While this may be suitable for some applications, it can be resource-intensive and unnecessary for others. Consider making the polling interval configurable or implementing a more efficient mechanism, such as WebSocket-based notifications.
- The fetchFriendRequests and fetchUsernames functions contain similar logic for making API requests and handling responses. Consider refactoring these functions to reduce code duplication and improve readability.